pkg/apis/keycloak/v1alpha1/keycloak_types.go

```diff
@@ -101,6 +101,12 @@ type KeycloakDeploymentSpec struct {
 	// the deployment to behave in an unpredictable fashion. Please use with care.
 	// +optional
 	Experimental ExperimentalSpec `json:"experimental,omitempty"`
+	// Toleration for the Pods when using taints and toleration on nodes
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+	// NodeSelector to constrain pods to run on particular set of Node(s)
+	// +optional
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
```

Comment:
Thinking about this more, this should actually go into `Experimental` where we already `Affinity` that is related to it.
https://github.com/keycloak/keycloak-operator/blob/1b9a02eb2417e9124ebbf4e9f37ec415d25c6b34/pkg/apis/keycloak/v1alpha1/keycloak_types.go#L133